src/main/java/com/elearningbackend/controller/BaseController.java

```diff
@@ -0,0 +1,48 @@
+package com.elearningbackend.controller;
+
+import com.elearningbackend.dto.CurrentUser;
+import com.elearningbackend.dto.UserDto;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import java.util.Currency;
+
+public class BaseController {
+
+    protected Authentication authentication;
+    protected CurrentUser currentUser;
+
+    /**
+     * Method get Current User from Context
+     * Can't call this method in Contructor because Context is null.
+     * Each Action in Controller, Call this method to get Current User
+     * @return UserDTO - CurrentUser
+     */
+    protected CurrentUser getCurrentUser(){
+        authentication = SecurityContextHolder.getContext().getAuthentication();
+        return (CurrentUser) authentication.getPrincipal();
```

Comment:
Như thế này thì các controller con phải làm cái việc CurrentUser currentUser = getCurrentUser(); 
đúng ko?

Reply:
Để nó toàn cục có đc ko em. Khi cần chỉ cần gọi đến curentUser luôn

Reply:
toàn cục thì vẫn được, nhưng phải khởi tạo httpcontext thì mới lấy được giá trị, bình thường khi vào strong controller thì context đã được khởi tạo, từ controller trở đi là gọi được. còn trong contructor của controller httpcontext nó vẫn sẽ null.

Reply:
E thử tnay xem đc k
Protected currentuser currentuser = getcurrentuser()

Reply:
cái này không được đâu, vì khi khởi tạo class sinh ra các filed mà thực hiện luôn hàm này thì bị lỗi null khi complier rồi. Bên C# làm jwt e cũng phải gọi hàm để convert ra user lúc cần lấy.

Reply:
Ok. Thế mình chấp nhận dùng như vậy đi. E check out file readme kia ra nhé. Xong a merge